src/loader/cmi.ml

```diff
@@ -504,8 +614,8 @@ and read_object env fi nm =
         let present_fields =
           List.fold_right
             (fun (n, k, t) l ->
-               match Btype.field_kind_repr k with
-               | Fpresent -> (n, t) :: l
+               match field_kind_repr k with
+               | f when f = field_public -> (n, t) :: l
```

Comment:
I think in this particular case it’s a bit better to avoid equality
```suggestion
#if OCAML_VERSION >= (4, 14, 0)
               | Types.Fpublic ->
#else
               | Types. Fpresent ->
#endif
                 (n, t) :: l
```